### DISCUSSION TITLE
Trying to make a function for suitability

### DISCUSSION THREAD

**Discussion by annaposlednik** [earthlab-education/Earth-Analytics-AY24#658] [Q&A]
Hello I have a bunch of data arrays which all start with one of my two site names. I was wondering if there's a way to create a function where I tell the function what site and year I want and for it to multiply across the respective data arrays. This is what I have so far, but I know its not right and I'm not sure if this is even possible to do. Any insights would be greatly appreciated! 
def suitability(site, year):
    """
    This function uses fuzzy logic to calculate the habitat suitability for Indiangrass
    by multiplying together pixels of elevation, slope, soil pH, and precipitation.  
    
    Parameters
    ----------
    
    site : 
        The grassland site, either "Pawnee" or "Comanche" 

    year: int
        The year for projected annual total precipitation.
    
    Returns
    -------
    DataArray
        A DataArray of habitat suitability for Indiangrass. 
    """
    
    precip = f'{site}_precip{year}_grid' > 400
    elev = f'{site}_elev_da' < 1700
    pH = f'{site}_pH_da_grid' > 4.8
    slope = f'{site}_slope_da_grid' < 20
    suitability_da = pH * slope * elev * precip
    
    
    return suitability_da

**Comment by akpost21** [comment]
@eculler can you help with this?

**Reply by eculler** [reply]
@annaposlednik this is possible but would be a bit unusual. You should look into the eval() function if you want to evaluate strings as code. 

Typically in Python, we would do something more like an object-oriented approach. This allows you to save values into an object, and then run the object's methods which reference those values -- objects can get their own attributes using the keyword self inside the class definition.
